### ISSUE TITLE
'NoMerge' and 'Deactivate' improvements

### ISSUE THREAD

**Issue by lojack5** [wrye-bash/wrye-bash#51]
[sourceforge](https://sourceforge.net/p/oblivionworks/enhancements/202/)

Currently, the only way for a mergeable mod to be separate from the bashed patch is to add NoMerge to the tags list. However, the NoMerge tag currently implies the Deactivate tag. This can be confusing and problematic for an end user, especially if they are unfamiliar with Bash, or if they are trying to track down a problem in the load order. I would like to suggest a tag that, like NoMerge, prohibits the records from transferring into the Bashed Patch, but also allows the mod to remain active if it was to begin with.

**Comment by GandaG** [comment]
GandaG/wrye-bash@31cdac4d re-enables deactivated NoMerge tagged mods once the bashed patch is done, appreciate it if anyone could review this. (@Utumno ? Whenever you have the time)

**Comment by Utumno** [comment]
Tried to review this but didn't quite get it - not sure at which point those mods should be reenabled

**Comment by GandaG** [comment]
What do you mean? They're reenabled right before the end, no?

**Comment by Utumno** [comment]
I don't think but I need a drink :P - after show modal and this is probably before the patch is built

**Comment by Utumno** [comment]
In `basher.patcher_dialog.PatchDialog#PatchExecute` - note the first line is `self.EndModalOK()` so at that point the dialog is ended and the lines: 

```
        with PatchDialog(self.window, self._selected_info, self.doCBash,
                         importConfig) as patchDialog:
            patchDialog.ShowModal()
        # lines below
        for mod in self.mods_to_reselect:
            bosh.modInfos.lo_activate(mod)
        bosh.modInfos.cached_lo_save_active()
        self.window.RefreshUI(refreshSaves=True)
        return self._selected_item
```

will run just after the dialog closes so before the build I'm affraid - should be moved in the finally clause of basher.patcher_dialog.PatchDialog#PatchExecute

I may be wrong hey - haven't looked at this part of the code for a while :)

When you get round to it could you please add an explanation here of what you are trying to achieve - will edit the commit message with that

Thanks! (and all the best with your exams !)

**Comment by Utumno** [comment]
Hmmm scratch this - it is run after `basher.patcher_dialog.PatchDialog#PatchExecute` - still maybe better to move there for clarity. Will attempt a refactoring

**Comment by Utumno** [comment]
I spent some time trying to refactor and kept hitting the various issues with that code - I would like to have some sort of finally in there but there are many flow paths in there and got no satisfactory solution

**Comment by Utumno** [comment]
Plus the distinction itself here could use some better help texts:

https://github.com/wrye-bash/wrye-bash/blob/d69f3e82c3afe4b6461b6ccce49a210bedd28dd6/Mopy/bash/basher/mod_links.py#L1018-L1043

So not sure what the diff between noMerge, deactivate etc is

(@Infernio too)

**Comment by Infernio** [comment]
See #161 for links and discussions about the behavior on all this. This seems like a big mess, with some mods relying on `NoMerge` actually deactivating mods, while other mods misunderstood what the tag does and require the plugin to remain active.

If we go the route of making `NoMerge` keep plugins active, we could make the combination of `NoMerge` and `Deactivate` restore the old behavior - the downside is that that would require a lot of LOOT/BOSS masterlist changes (525 `NoMerge` usages in the [LOOT masterlist](https://github.com/loot/oblivion/blob/v0.14/masterlist.yaml), 543 in the [BOSS masterlist](https://github.com/boss-developers/oblivion/blob/master/masterlist.txt)).

**Comment by Utumno** [comment]
Thanks @Infernio - if you can lend a coding hand it'd be great. We don't really want any user visible tag updates really so maybe drop it and instead create a new tag? Not sure at all

**Comment by Utumno** [comment]
For now I pushed https://github.com/wrye-bash/wrye-bash/commit/00ec305efa9064945d4f88a5035c9244d7b526be which has small fixups and should be squashed to the commit by @GandaG if we go down that path - gosh it was a good 5 hours, I did need a lazy Sunday to review.

@Infernio I assigned to you #161 - maybe a good time to clear that mess - we are approaching the patcher, from the periphery

**Comment by Infernio** [comment]
Should have been closed in 13a52d2f91ebef85901a0218fc0fb0576e8457ae - as mentioned above, further discussion about this mess is 308 territory and will occur in #161.
